### PR TITLE
#163519149 Implement email and in-app notification feature

### PR DESCRIPTION
**Pull request by mcaleb808**
#### What does this PR do?
- Implement email and in-app notification feature

#### Description of Task to be completed?
- write unit tests
- create notification's model and migration
- implement notification for new article from people you follow
- implement notification for new interaction on article you liked or commented on
- implement new follower notification
- implement notification endpoints
- implement disable and enable notification option

#### How should this be manually tested?
On this branch
- Run `npm run migrate:seed`

- Run `npm run test`

- Run `npm run dev`

With postman send post requests to these endpoints:
- All notifications `GET: api/v1/notications`

- One notification `GET: api/v1/notification/:notificationId`

- Mark all notifications as read `PUT: api/v1/notification/markasread`

- Enable notification `PUT: api/v1/notification/enable`

- disable notification `PUT: api/v1/notification/disable`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#163519149

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

[finishes #163519149]

### CODE REVIEW

**Review by oesukam — CHANGES_REQUESTED** [review]
@mcaleb808 
Try to add more tests for coveralls

**Review by NkFab — APPROVED** [review]
Good work Caleb.

**Comment by mcaleb808** [comment]
> Good work Caleb.

Thank you @NkFab

**Review by oesukam — APPROVED** [review]

File: __tests__/routes/article.test.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import { Op } from 'sequelize';
 import bcrypt from 'bcrypt';
 import app from '../../app';
 import { urlPrefix } from '../mocks/variables.json';
-import { User, Article, Favorite, Report } from '../../database/models';
+import { User, Article, Favorite, Report, Favorite } from '../../database/models';
```
[the remark this answers is not in the document]
Comment (oesukam) [reply __tests__/routes/article.test.js]: @mcaleb808 You have imported the Favorite model two times

[the remark this answers is not in the document]
Comment (mcaleb808) [reply __tests__/routes/article.test.js]: @oesukam I fixed that

**Pull request merged**